sycl/include/sycl/ext/oneapi/experimental/syclbin_kernel_bundle.hpp

```diff
@@ -24,6 +25,13 @@
 
 namespace sycl {
 inline namespace _V1 {
+
+namespace detail {
+__SYCL_EXPORT std::shared_ptr<detail::kernel_bundle_impl>
+link_impl(const std::vector<kernel_bundle<bundle_state::object>> &ObjectBundles,
```

Comment:
I do not like the idea of using the specific container (`std::vector` in that case) when we need to pass the range of objects to the ABI entry point.

Reply:
Good point! It was copy-pasted from the existing link_impl without further thought, but I agree that passing ranges here is better. I decided to go with C-style pointer + size arguments to avoid `sycl::span` becoming a part of the ABI for this. It is converted to a span in the library however, to simplify the usage.

Reply:
I'm not sure. We could, but I prefer C-style arrays to make the ABI breaking surface smaller once we have the actual `std::span`.